extensions/git/src/git.ts

```diff
@@ -1300,8 +1302,16 @@ export class Repository {
 		}
 	}
 
-	async commit(message: string, opts: CommitOptions = Object.create(null)): Promise<void> {
-		const args = ['commit', '--quiet', '--allow-empty-message', '--file', '-'];
+	async commit(message?: string, opts: CommitOptions = Object.create(null)): Promise<void> {
+		const args = ['commit', '--quiet'];
+		const options: SpawnOptions = {};
+
+		if (!opts.useEditor || message) {
```

Comment:
If a message is supplied, it should still be used for the commit. This allows to use the input box for quick commits, although `git.useEditorToCommit` is enabled.